extensions/amp-pinterest/0.1/util.js

```diff
@@ -39,20 +39,17 @@ function log(queryParams) {
 }
 
 /**
- * Strip data from string
+ * Pinterest provides text HTML encoded. This utility transforms it back
+ * into UTF-8 text.
  * @param {string} str - the string to filter
  * @return {string}
  */
 function filter(str) {
-  let decoded, ret;
-  decoded = '';
-  ret = '';
   try {
-    decoded = decodeURIComponent(str);
-  } catch (e) {}
-  ret = decoded.replace(/</g, '&lt;');
-  ret = ret.replace(/>/g, '&gt;');
-  return ret;
+    return new DOMParser().parseFromString(str, 'text/html').body.textContent;
```

Comment:
Are we concerned about performance here? Closure has a lighter weight (perf, but larger file size) approach: https://github.com/google/closure-library/blob/8d4f8222eb3729e5aa50c07e82393b15eaf1ab1a/closure/goog/string/string.js#L605-L649

Reply:
No, I think we don't really care. In practice this will be very rare operations.